ucp/benchmarks/utils.py

```diff
@@ -0,0 +1,353 @@
+        else:
+            fp = open(server_file, mode="w")
+        json.dump({"address": ucx_api.get_address(), "port": lf.port}, fp)
+        fp.close()
+
+        while len(results) != n_workers:
+            await asyncio.sleep(0.1)
+
+        return results
+
+    loop = asyncio.new_event_loop()
+    ret = loop.run_until_complete(run())
+    for rank in range(n_workers):
+        if q is None:
+            print(ret[rank])
+        else:
+            q.put(ret[rank])
+
+
+def _run_cluster_server(
+    server_file,
+    n_workers,
+    ucx_options_list=None,
+):
+    """
+    Create a server that synchronizes workers.
+
+    The server will wait for all `n_workers` to connect and communicate their
+    endpoint information, then send the aggregate information to all workers
+    so that they will create endpoints to each other, in a fully-connected
+    network. Each worker will then communicate its result back to the scheduler
+    which will return that result back to the caller.
+
+    Parameters
+    ----------
+    server_file: str or None
+        A string containing the path to a file that will be populated to contain
+        the address and port of the server, or `None` to print that information
+        to stdout.
+    num_workers : int
+        Number of workers in the entire network, required to infer when all
+        workers have connected and completed.
+    ucx_options_list: list of dict
+        Options to pass to UCX when initializing workers, one for each worker.
+
+    Returns
+    -------
+    return : tuple
+        A tuple with two elements: the process spawned and a queue where results
+        will eventually be stored.
+    """
+    q = mp.Queue()
+    p = mp.Process(
+        target=_server_process,
+        args=(
+            q,
+            server_file,
+            n_workers,
+            ucx_options_list,
+        ),
+    )
+    p.start()
+    return p, q
+
+
+def run_cluster_server(
+    server_file,
+    n_workers,
+    ucx_options_list=None,
+):
+    """
+    Blocking version of `_run_cluster_server()`.
+
+    Provides same behavior as `_run_cluster_server()`, except that it will join
+    processes and thus cause the function to be blocking. It will also combine
+    the queue as a list with results for each worker in the `[0..rank-1)` range.
+    """
+    p, q = _run_cluster_server(
+        server_file=server_file,
+        n_workers=n_workers,
+        ucx_options_list=ucx_options_list,
+    )
+
+    # Joining the process if the queue is too large (reproducible for more than
+    # 32 workers) causes the process to hang. We join the queue results in a
+    # list and return the list instead.
+    ret = [q.get() for i in range(n_workers)]
+
+    p.join()
+    assert not p.exitcode
+
+    return ret
+
+
+def _worker_process(
+    queue,
+    server_info,
+    num_node_workers,
+    rank,
+    ucx_options_list,
+    ensure_cuda_device,
+    func,
+    args,
+):
+    if ensure_cuda_device is True:
```

Comment:
Yes, my specific reason is to avoid accidental matches. The fact that `[]`, `()`, `False`, `None` all evaluate to `False` is a terrible pattern. Added to that, the fact that in Python we can pass any other value by mistake to an argument without realizing it and it will evaluate to `True` without the check against `True` itself, is also horrible. Therefore, I prefer to be explicit and make sure I'm testing against the expected value.